hands_on/sync_upstream.py

```diff
@@ -0,0 +1,44 @@
+import os
+import sys
+import shutil
+
+from exercise_utils.github_cli import (
+    get_github_username,
+    fork_repo,
+    clone_repo,
+    has_repo,
+)
+
+__requires_git__ = True
+__requires_github__ = True
+
+
+TARGET_REPO = "git-mastery/samplerepo-finances"
+LOCAL_DIR = "samplerepo-finances"
+
+
+def _get_full_repo_name(verbose: bool) -> str:
+    username = get_github_username(verbose)
+    return f"{username}/{LOCAL_DIR}"
+
+
+def download(verbose: bool):
+    full_repo_name = _get_full_repo_name(verbose)
+
+    if has_repo(full_repo_name, is_fork=True, verbose=verbose):
+        print(
+            f"\nRepository 'https://github.com/{full_repo_name}' already exists on GitHub.\n"
+            "Please delete the existing fork before proceeding."
+        )
+        sys.exit(1)
```

Comment:
As discussed in this issue, we are opting to enforce that students always have the `delete_repo` scope through `__requires_github__ = True`. So this should be a direct `delete_repo` call: https://github.com/git-mastery/exercises/issues/119